src/wasm-binary.h

```diff
@@ -1430,7 +1430,7 @@ class WasmBinaryBuilder {
   MixedArena& allocator;
   const std::vector<char>& input;
   std::istream* sourceMap;
-  std::pair<uint32_t, Function::DebugLocation> nextDebugLocation;
+  std::tuple<uint32_t, uint32_t, Function::DebugLocation> nextDebugLocation; // available pos, previous pos, next debug location
```

Comment:
I think it would be good to use a struct here. Something like
```cpp
struct NextDebugLocation {
  uint32_t available;
  uint32_t previous;
  uint32_t next;
};
```

Reply:
Hi Alon,
I updated the fix according to your comments, and for the next step (add `lit` test to main test suite), I donot know how to begin the work, can you give some advice? thank you.

Reply:
For the test, just create a new file at `tests/lit/source-map.wast`, paste the test contents from [here](https://github.com/WebAssembly/binaryen/pull/5504#pullrequestreview-1309749525) into that file, and add it using `git add`, then commit it. It should just work.

Reply:
The lit test just passed as well:
```
/binaryen$ python build/bin/binaryen-lit -vv test/lit/source-map.wast 
-- Testing: 1 tests, 1 workers --
PASS: Binaryen lit tests :: source-map.wast (1 of 1)

Testing Time: 0.06s
  Passed: 1
```

Reply:
Hi Alon,
the test has been added, please kindly review, thank you.